### ISSUE TITLE
Visible range isn't updated when measurements change

### ISSUE THREAD

**Issue by piecyk** [tannerlinsley/react-virtual#112]
Hi 

noticed that when measurements change after initial render, visible range isn't updated. 
Range is updated on initial render, and next on every scroll.

Maybe let's add isScrolling on state and have an extra effect 

```tsx
useIsomorphicLayoutEffect(() => {
  if (!isScrolling && latestRef.current.scrollOffset !== undefined) {
    setRange(prevRange => calculateRange(latestRef.current, prevRange))
  }
}, [isScrolling, measurements])
```

**Comment by piecyk** [comment]
Reproducible example https://codesandbox.io/s/react-virtualissues112-eiqs0

**Comment by piecyk** [comment]
This is pretty critical bug in context of dynamic measurements.

**Comment by piecyk** [comment]
Closed by #166
